File: macos/Scanner/Sources/Scanner/Rules/AESMCP005.swift
public struct AESMCP005: Rule {
    public let id = "AES-MCP-005"
    public let severity: Severity = .high

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        let browserMarkers = [
            "playwright",
            "puppeteer",
            "browser-use",
            "selenium",
            "@playwright/mcp",
            "playwright-mcp"
        ]

        return facts.mcpServers.compactMap { server in
            guard !server.disabled else {
                return nil
            }

            let values = server.searchableValues.map { $0.lowercased() }
            guard values.contains(where: { value in browserMarkers.contains(where: value.contains) }) else {
                return nil
            }
            return finding(app: server.appId, serverName: server.name, affectedPath: server.configPath)
        }
    }
}

File: windows/Scanner/Rules/AESMCP005.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESMCP005 : IRule
{
    private static readonly string[] BrowserMarkers =
    [
        "playwright",
        "puppeteer",
        "browser-use",
        "selenium",
        "@playwright/mcp",
        "playwright-mcp"
    ];

    public string Id => "AES-MCP-005";
    public Severity Severity => Severity.High;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.McpServers
            .Where(server => !server.Disabled)
            .Where(server => server.SearchableValues()
                .Select(value => value.ToLowerInvariant())
                .Any(value => BrowserMarkers.Any(value.Contains)))
            .Select(server => this.Finding(server.AppId, server.Name, affectedPath: server.ConfigPath))
            .ToArray();
}
